odoo/cli/obfuscate.py

```diff
@@ -0,0 +1,249 @@
+        group.add_option('--exclude', dest="exclude", default=False, help="List of table.columns to exclude from obfuscate/unobfuscate: table1.column1,table2.column1,table2.column2")
+        group.add_option('--file', dest="file", default=False, help="File containing the list of table.columns to obfuscate/unobfuscate")
+        group.add_option('--unobfuscate', action='store_true', default=False)
+        group.add_option('--allfields', action='store_true', default=False, help="Used in unobfuscate mode, try to unobfuscate all fields. Cannot be used in obfuscate mode. Slower than specifying fields.")
+        group.add_option('--vacuum', action='store_true', default=False, help="Vacuum database after unobfuscating")
+        group.add_option('--pertablecommit', action='store_true', default=False, help="Commit after each table instead of a big transaction")
+
+        parser.add_option_group(group)
+        if not cmdargs:
+            sys.exit(parser.print_help())
+
+        try:
+            opt = odoo.tools.config.parse_config(cmdargs)
+            if not opt.pwd:
+                _logger.error("--pwd is required")
+                sys.exit("ERROR: --pwd is required")
+            if opt.allfields and not opt.unobfuscate:
+                _logger.error("--allfields can only be used in unobfuscate mode")
+                sys.exit("ERROR: --allfields can only be used in unobfuscate mode")
+            self.dbname = odoo.tools.config['db_name']
+            self.registry = odoo.registry(self.dbname)
+            with self.registry.cursor() as cr:
+                self.cr = cr
+                self.begin()
+                if self.check_pwd(opt.pwd):
+                    fields = [
+                            ('mail_tracking_value', 'old_value_char'),
+                            ('mail_tracking_value', 'old_value_text'),
+                            ('mail_tracking_value', 'new_value_char'),
+                            ('mail_tracking_value', 'new_value_text'),
+                            ('res_partner', 'name'),
+                            ('res_partner', 'display_name'),
+                            ('res_partner', 'email'),
+                            ('res_partner', 'phone'),
+                            ('res_partner', 'mobile'),
+                            ('res_partner', 'street'),
+                            ('res_partner', 'street2'),
+                            ('res_partner', 'city'),
+                            ('res_partner', 'zip'),
+                            ('res_partner', 'vat'),
+                            ('res_partner', 'website'),
+                            ('res_country', 'name'),
+                            ('mail_message', 'subject'),
+                            ('mail_message', 'email_from'),
+                            ('mail_message', 'reply_to'),
+                            ('mail_message', 'body'),
+                            ('crm_lead', 'title'),
```

Comment:
> many2one field that can not be obfuscated : completely expected. the check_field method is checking it.

I didn't make myself understood :  ``crm_lead.title`` is not a ``field.Char``. (the field is bad named). It is a many2one. As a result, that line is incorrect, generate a warning, and the value will not be obfuscated.

So, I suggest that change : 

```suggestion
```

Reply:
the aim of adding this field is ensuring the `check_field` method correctly remove the field.

Reply:
I'm not sure to understand.

A) if the objective of this line is to test the removal of this field for this draft PR to make test and then it will be removed before the merge of the PR : :+1: 
B) if the objective is to keep this line once the PR is merged : :-1: . The current design generate a warning message when running the obfuscate command. It's going to be confusing for the user.